applications/directory/src/components/CreateProfileForm.js

```diff
@@ -31,20 +31,78 @@ const categoriesDefaultValue = 'Dancer';
 const tagsDefaultValue = 'Education';
 const skillsDefaultValue = 'Carpentry';
 
-const initialValues = {
-  firstName: '',
-  lastName: '',
-  email: '',
-  city: '',
-  website: { checked: true, name: 'website', url: '' },
-  behance: { checked: false, name: 'behance', url: '' },
-  other: { checked: false, name: 'other', url: '' },
-  description: '',
-  categories: [categoriesDefaultValue],
-  tags: [tagsDefaultValue],
-  skills: [skillsDefaultValue],
-  files: [],
-  subscribedToNewsletter: 'yes'
+// convert artist data from backend to the values to be excepted by the form's "initial values"
+const parseArtist = (artist) => {
+  // check if artist is an empty object
+  if (Object.keys(artist).length === 0) {
+    return false;
+  }
+
+  const {
+    firstName,
+    lastName,
+    email,
+    city,
+    description,
+    categories,
+    social,
+    tags,
+    skills,
+    images
+  } = artist;
+
+  // parse images array to shape them into the form Upload component uses them to upload new images
+  const parsedImages = images.map((image) => {
+    const imageName = image.split('/').at(-1); // returns last array value which is a file name
+    return {
+      fileName: imageName,
+      file: { name: imageName, preview: image },
+      uploadError: undefined,
+      signedUrlError: undefined,
+      uploaded: true,
+      loading: false
+    };
+  });
+
+  // convert social to object
+  const parsedExistingSocial = social.reduce((list, item) => {
+    return {
+      ...list,
+      [item.name]: {
+        name: item.name,
+        url: item.url,
+        checked: Boolean(item.url)
+      }
+    };
+  }, {});
+
+  // create list of social like initial values for the form
+  const socialInitialValues = {
+    website: { checked: true, name: 'website', url: '' },
```

Comment:
Is `checked` going to be saved in the database? e.g., `artist.social.website.checked` -- will this be in the database?

If so, why?

Reply:
Basically I'm wondering: should we save it to the database but let them "turn it off?" Or should we delete it if they turn it off?

Reply:
I know why i did it but I just now realised what was the problem and how to solve it. No, we do not need the 'checked' value. I forgot that i could make a deep array copy so social array manipulation would not effect original value of social items (without the deep copy it gave me a warning about changing from controlled input to uncontrolled in the form). I will change createArtist route as well (i was removing this 'checked" property there). 
Done